src/chain/mod.rs

```diff
@@ -35,7 +37,8 @@ const SHUTDOWN_TIMEOUT: Duration = Duration::from_millis(120000);
 /// RPC server handle
 #[derive(Clone, Debug)]
 pub struct ChainManager {
-    pub tx: tokio::sync::mpsc::Sender<ChainRequest>,
+    pub tx: mpsc::Sender<ChainRequest>,
+    connected_rpcs: Arc<RwLock<Vec<RpcInfo>>>,
```

Comment:
```suggestion
```
@Slesarew decided to use channels instead of locks everywhere. With this approach, all variables should be declared in an initialization function, not in a struct, like there:
https://github.com/Alzymologist/Kalatori-backend/blob/fe135a3c83477d7f7e98c6773b5e2e704a24e799/src/chain/mod.rs#L56-L58
https://github.com/Alzymologist/Kalatori-backend/blob/fe135a3c83477d7f7e98c6773b5e2e704a24e799/src/state.rs#L65-L76
I'm against this approach, and plan to refactor it in locks later, but for now I suggest to abide by at least one approach for every module instead of mixing them.

Reply:
Addressed